Sources/AppAuth/OKTAuthorizationService.m

```diff
@@ -610,34 +615,64 @@ + (void)performTokenRequest:(OKTTokenRequest *)request
 
       // OpenID Connect Core Section 3.1.3.7. rule #9
       // Validates that the current time is before the expiry time.
-      NSTimeInterval expiresAtDifference = [idToken.expiresAt timeIntervalSinceNow];
-      if (expiresAtDifference < 0) {
-        NSError *invalidIDToken =
-            [OKTErrorUtilities errorWithCode:OKTErrorCodeIDTokenFailedValidationError
-                             underlyingError:nil
-                                 description:@"ID Token expired"];
-        dispatch_async(dispatch_get_main_queue(), ^{
-          callback(nil, invalidIDToken);
-        });
-        return;
+      if (validator != nil) {
+          if ([validator isExpired:idToken.expiresAt]) {
+              NSError *invalidIDToken =
+                  [OKTErrorUtilities errorWithCode:OKTErrorCodeIDTokenFailedValidationError
+                                   underlyingError:nil
+                                       description:@"ID Token expired"];
+              dispatch_async(dispatch_get_main_queue(), ^{
+                callback(nil, invalidIDToken);
+              });
+              return;
+          }
+      } else {
+          NSTimeInterval expiresAtDifference = [idToken.expiresAt timeIntervalSinceNow];
+          if (expiresAtDifference < 0) {
+            NSError *invalidIDToken =
+                [OKTErrorUtilities errorWithCode:OKTErrorCodeIDTokenFailedValidationError
+                                 underlyingError:nil
+                                     description:@"ID Token expired"];
+            dispatch_async(dispatch_get_main_queue(), ^{
+              callback(nil, invalidIDToken);
+            });
+            return;
+          }
```

Comment:
For the default behaviour, I'd recommend we create a token validator and default to using that, moving this default logic into that other class.  This should provide several benefits:

1. The expiration and issuedAtValid checks can be centralized in one place.
2. It will be easier to unit test the time check logic
3. It will provide an example to developers as to how they should go about implementing their own
4. It will simplify the code, meaning there's only a single code-path to follow, as opposed to the if/else clauses here.

As a result, this class will always have a validator object, and will simply assign the default instance if `nil` is supplied.